restructuredtext_lint/ignore.py

```diff
@@ -0,0 +1,79 @@
+import contextlib
+
+from docutils.parsers import rst
+from docutils.parsers.rst import directives as rst_directives
+from docutils.parsers.rst import roles as rst_roles
+
+
+class _IgnoredDirective(rst.Directive):
+    """Stub for unknown directives.
+
+    See: http://repo.or.cz/w/docutils.git/blob/1976ba91eff979abc3e13e5d8cb68324833af6a0:/docutils/parsers/rst/__init__.py#l194  # noqa
+
+    See: https://github.com/myint/rstcheck/blob/v1.1.1/rstcheck.py#L217
+    """
+    has_content = True
+
+    def run(self):
+        return []
+
+
+def _ignore_role(name, rawtext, text, lineno, inliner,
+                 options=None, content=None):
+    """Stub for unknown roles.
+
+    See: http://repo.or.cz/w/docutils.git/blob/1976ba91eff979abc3e13e5d8cb68324833af6a0:/docutils/parsers/rst/roles.py
+
+    See: https://github.com/myint/rstcheck/blob/v1.1.1/rstcheck.py#L228
+    """
+    return ([], [])
+
+
+def register_ignores(directives, roles):
+    """Register ignoreable sphinx directives & roles.
+
+    :param list directives: directives to ignore
+    :param list roles: roles to ignore
+
+    :rtype (list, list): tuple of directives and roles.
+    """
+    for directive in directives:
+        rst_directives.register_directive(directive, _IgnoredDirective)
+    for role in roles:
+        rst_roles.register_local_role(role, _ignore_role)
+    return (directives, roles)
+
+
+def unregister_ignores(registered_directives, registered_roles):
+    """Unregister previously registered sphinx directives & roles.
+
+    :param list registered_directives: directives to unregister
+    :param list registered_roles: roles to unregister
+    """
+    # TODO: this is a hack into the docutils rst registries, there doesn't
+    # appear to be any other way to get at these...
+    all_directives = getattr(rst_directives, '_directives', {})
```

Comment:
More accurately, these are `registered_directives` because they are actually properties of `rst_directives.